src/main/java/com/anaplan/client/AnaplanService.java

```diff
@@ -26,9 +40,9 @@
 
   /** Interface for the CDF plugin that use Anaplan service. */
   public interface AnaplanConfig {
-    String getServiceLocation();
+    URI getServiceLocation() throws URISyntaxException;
```

Comment:
this shouldn't throw URISyntaxException.  Parsing the String into a URI should happen before the AnaplanConfig is created, as we don't want callers to have to handle the exception.

Reply:
Thanks for this point! Here comes the limitation ->

The AnaplanConfig will be implemented by PluginConfig. And the serviceLocation is one of the PluginConfig fields which has to be a primitive type String. And as I know the PluginConfig fields are populated by reflection rather than a constructor. Therefore, this getter is the first chance to convert the String serviceLocation to URI.. Or we rolled back to the original solution, throw the URISyntaxException in the setter (without throwing a RuntimeException), does it make more sense?

Could you please advice on this? Thanks!

Reply:
I see. In that case it would be cleaner not to have the PluginConfig implement this interface, but to have an extra step where the AnaplanConfig is created from the PluginConfig (with the AnaplanConfig constructor taking URIs for these locations and not Strings).

Reply:
Thanks! That's a better idea! The solution is update!